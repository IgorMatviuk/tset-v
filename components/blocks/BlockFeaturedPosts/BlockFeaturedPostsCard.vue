<template>
  <div class="post-card">
    <div class="post-card__row">
      <div class="post-card__preview">
        <div class="post-card__img-wrapper">
          <ImageLoader :negative="true">
            <picture>
              <source
                :data-srcset="post.images.extraLarge | webp"
                srcset="/img/stub.png"
                media="(min-width: 1500px)"
                type="image/webp"
              />
              <source
                :data-srcset="post.images.extraLarge"
                srcset="/img/stub.png"
                media="(min-width: 1500px)"
                type="image/jpeg"
              />
              <source
                :data-srcset="post.images.large | webp"
                srcset="/img/stub.png"
                media="(min-width: 1200px)"
                type="image/webp"
              />
              <source
                :data-srcset="post.images.large"
                srcset="/img/stub.png"
                media="(min-width: 1200px)"
                type="image/jpeg"
              />
              <source
                :data-srcset="post.images.square | webp"
                srcset="/img/stub.png"
                media="(min-width: 992px)"
                type="image/webp"
              />
              <source
                :data-srcset="post.images.square"
                srcset="/img/stub.png"
                media="(min-width: 992px)"
                type="image/jpeg"
              />
              <source
                :data-srcset="post.images.small | webp"
                type="image/webp"
              />
              <img
                :data-src="post.images.small"
                :alt="post.title"
                :title="post.title"
                src="/img/stub.png"
                class="post-card__img lazyload img-fade"
                loading="lazy"
              />
            </picture>
          </ImageLoader>
          <div class="post-card__img-overlay"></div>
          <div class="post-card__button-group">
            <BButton
              to="/"
              variant="outline-light"
              class="post-card__button is-blog"
            >
              в блог
            </BButton>
          </div>
        </div>
      </div>
      <article class="post-card__content">
        <div class="post-card__date">{{ post.pubDate }}</div>
        <!-- TODO: remove views -->
        <!-- <div class="post-card__views">
          <svg-icon
            name="views"
            class="post-card__views-icon"
            width="15"
            height="15"
          />
          {{ post.views }}
        </div> -->
        <h3 class="post-card__title">{{ post.title }}</h3>
        <p class="post-card__post">{{ post.post }}</p>
        <!-- TODO: one post url -->
        <MoreLink :is-black="true" url="some-article" class="post-card__link">
          {{ $t('buttons.readMore') }}
        </MoreLink>
      </article>
    </div>
  </div>
</template>

<script>
import ImageLoader from '@/components/shared/ImageLoader'
import MoreLink from '@/components/shared/MoreLink'

import imageNextGen from '@/mixins/image-next-gen'

export default {
  components: {
    ImageLoader,
    MoreLink
  },
  mixins: [imageNextGen],
  props: {
    post: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss">
.post-card {
  &__row {
    position: relative;
    display: flex;
    align-items: flex-start;
    margin-right: 0;
    margin-left: 0;
  }

  &__preview {
    flex-shrink: 0;
    width: 330px;
  }

  &__img-wrapper {
    position: relative;
    height: 1px;
    padding-bottom: $preview-aspect-ratio;
  }

  &__img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: center;
  }

  &__img-overlay {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: $black-light-1;
    opacity: 0;
    transition: opacity 0.5s;
  }

  &:hover &__img-overlay {
    opacity: 1;
  }

  &__button-group {
    position: absolute;
    top: calc(50% - 20px);
    left: calc(50% - 97px);
    opacity: 0;
    transition: transform 0.5s, opacity 0.5s;
    transform: translateY(25px);
  }

  &:hover &__button-group {
    opacity: 1;
    transform: translateY(0);
  }

  &__button {
    min-width: 180px;
    font-size: 14px;
  }

  &__content {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 330px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 345px;
    height: 100%;
    padding-right: 12px;
    padding-left: 30px;
  }

  &__date,
  &__views {
    font-size: 14px;
    line-height: 1;
  }

  &__date {
    width: 100%;
  }

  &__views {
    display: flex;
    align-items: center;
    margin-top: -1px;
  }

  &__views-icon {
    width: 15px;
    height: 15px;
    margin-right: 5px;
    color: $gray-dark-1;
  }

  &__title {
    display: -webkit-box;
    height: 90px;
    margin-top: 20px;
    margin-bottom: 0;
    overflow: hidden;
    font-size: 24px;
    font-weight: 700;
    line-height: 1.25;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
  }

  &__post {
    display: -webkit-box;
    height: 281px;
    margin-top: 28px;
    margin-bottom: 0;
    overflow: hidden;
    font-size: 16px;
    line-height: 1.25;
    color: $text-color;
    -webkit-line-clamp: 14;
    -webkit-box-orient: vertical;
  }

  &__link {
    margin-top: auto;
  }

  @include media-breakpoint-down(xl) {
    &__preview {
      width: 270px;
    }

    &__button-group {
      left: calc(50% - 84px);
    }

    &__button {
      min-width: 150px;
      height: 33px;
      line-height: 31px;
    }

    &__content {
      left: 270px;
      width: 295px;
      padding-right: 0;
    }

    &__date {
      margin-right: 20px;
    }

    &__views-icon {
      margin-right: 5px;
    }

    &__title {
      height: 120px;
      margin-top: 15px;
      -webkit-line-clamp: 4;
    }

    &__post {
      height: 180px;
      margin-top: 28px;
      -webkit-line-clamp: 9;
    }
  }

  @include media-breakpoint-down(lg) {
    &__preview {
      width: 290px;
    }

    &__img-overlay {
      display: none;
    }

    &:hover &__img-overlay {
      opacity: 0;
      transform: none;
    }

    &__button-group {
      top: initial;
      bottom: 55px;
      left: calc(50% - 93px);
      opacity: 1;
      transform: none;
    }

    &:hover &__button-group {
      opacity: 1;
      transform: none;
    }

    &__button {
      min-width: 170px;
      height: 35px;
      line-height: 33px;

      &.is-post {
        margin-bottom: 0;
      }

      &.is-blog {
        display: none;
      }
    }

    &__content {
      left: 290px;
      width: unset;
      padding-right: 0;
    }

    &__views-icons {
      margin-right: 6px;
    }

    &__title {
      height: 60px;
      margin-top: 20px;
      -webkit-line-clamp: 2;
    }

    &__post {
      height: 220px;
      margin-top: 58px;
      -webkit-line-clamp: 11;
    }
  }

  @include media-breakpoint-down(md) {
    &__preview {
      width: 210px;
    }

    &__button-group {
      bottom: 40px;
      left: calc(50% - 90px);
    }

    &__content {
      left: 210px;
    }

    &__views-icon {
      margin-right: 8px;
    }

    &__title {
      height: 90px;
      -webkit-line-clamp: 3;
    }

    &__post {
      height: 120px;
      margin-top: 28px;
      -webkit-line-clamp: 6;
    }
  }

  @include media-breakpoint-down(sm) {
    &__row {
      flex-direction: column;
    }

    &__preview {
      width: calc(100% + 30px);
      margin-right: -15px;
      margin-left: -15px;
    }

    &__button-group {
      bottom: 37px;
      left: calc(50% - 105px);
    }

    &__button {
      min-width: 210px;
      font-size: 13px;
    }

    &__content {
      position: static;
      width: 100%;
      padding-left: 0;
      margin-top: 13px;
    }

    &__title {
      width: 100%;
      height: 60px;
      margin-top: 22px;
      font-size: 16px;
    }

    &__post {
      height: 55px;
      margin-top: 14px;
      font-size: 14px;
      line-height: 18px;
      -webkit-line-clamp: 3;
    }

    &__link {
      margin-top: 15px;
    }
  }
}
</style>
