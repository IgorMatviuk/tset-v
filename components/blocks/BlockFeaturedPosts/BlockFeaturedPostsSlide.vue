<template>
  <div class="swiper-slide posts__slide">
    <BlockFeaturedPostsCard :post="post" />
  </div>
</template>

<script>
import BlockFeaturedPostsCard from './BlockFeaturedPostsCard'

export default {
  components: {
    BlockFeaturedPostsCard
  },
  props: {
    post: {
      type: Object,
      required: true
    }
  }
}
</script>
