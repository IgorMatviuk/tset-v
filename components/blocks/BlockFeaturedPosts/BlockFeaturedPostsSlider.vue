<template>
  <div
    v-swiper:featuredPostsSlider="swiperOptions"
    :cleanup-styles-on-destroy="false"
    class="posts__slider"
  >
    <div class="swiper-wrapper posts__slider-wrapper">
      <BlockFeaturedPostsSlide
        v-for="post in posts"
        :key="post.id"
        :post="post"
      />
    </div>
    <SliderPagination class="posts__pagination" />
  </div>
</template>

<script>
import BlockFeaturedPostsSlide from './BlockFeaturedPostsSlide'
import SliderPagination from '@/components/shared/SliderPagination'

export default {
  components: {
    BlockFeaturedPostsSlide,
    SliderPagination
  },
  props: {
    posts: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      swiperOptions: {
        watchOverflow: true,
        slidesPerView: 1,
        spaceBetween: 50,
        navigation: {
          nextEl: '.post__next',
          prevEl: '.post__prev'
        },
        pagination: {
          el: '.posts__pagination',
          clickable: true
        },
        breakpoints: {
          1200: {
            slidesPerView: 2,
            spaceBetween: 30
          }
        }
      }
    }
  }
}
</script>
