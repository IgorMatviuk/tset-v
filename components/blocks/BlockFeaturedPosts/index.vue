<template>
  <div class="posts">
    <BContainer>
      <SectionHeader>
        <template v-slot:title>{{ title }}</template>
        <template v-slot:navigation>
          <div class="posts__navigation">
            <SliderArrow arrow-type="prev" class="post__prev" />
            <SliderArrow arrow-type="next" class="post__next" />
          </div>
          <MoreLink url="blog" class="posts__more">
            Весь блог
          </MoreLink>
        </template>
      </SectionHeader>
      <div class="posts__slider-container">
        <BlockFeaturedPostsSlider :posts="posts" />
      </div>
    </BContainer>
  </div>
</template>

<script>
import BlockFeaturedPostsSlider from './BlockFeaturedPostsSlider'
import SectionHeader from '@/components/shared/SectionHeader'
import SliderArrow from '@/components/shared/SliderArrow'
import MoreLink from '@/components/shared/MoreLink'

export default {
  components: {
    SectionHeader,
    BlockFeaturedPostsSlider,
    SliderArrow,
    MoreLink
  },
  props: {
    posts: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      title: 'блог'
    }
  }
}
</script>

<style lang="scss">
.posts {
  margin-top: 30px;

  &__navigation {
    position: relative;
    top: -3px;
    right: -5px;
    width: 115px;
  }

  &__more {
    display: none;
  }

  &__slide {
    flex: 0 0 690px;
    max-width: 690px;
    margin-right: 30px;
  }

  &__pagination {
    display: none;
  }

  @include media-breakpoint-down(xl) {
    margin-top: 34px;

    &__slide {
      flex: 0 0 570px;
      max-width: 570px;
      margin-right: 30px;
    }
  }

  @include media-breakpoint-down(lg) {
    margin-top: 14px;

    &__navigation {
      display: none;
    }

    &__more {
      display: block;
      padding-top: 4px;
      margin-right: -3px;
    }

    &__slider {
      padding-bottom: 52px;
    }

    &__slide {
      flex: 0 0 auto;
      max-width: 100%;
      margin-right: 0;
    }

    &__pagination {
      display: block;
    }
  }

  @include media-breakpoint-down(md) {
    margin-top: 9px;
  }

  @include media-breakpoint-down(sm) {
    margin-top: 14px;

    &__more {
      padding-top: 0;
    }

    &__slider-container {
      margin-right: -15px;
      margin-left: -15px;
    }

    &__slider {
      padding-right: 15px;
      padding-bottom: 54px;
      padding-left: 15px;
    }
  }
}
</style>
