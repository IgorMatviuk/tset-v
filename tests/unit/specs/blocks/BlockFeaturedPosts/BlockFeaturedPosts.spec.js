import { mount, createLocalVue } from '@vue/test-utils'
import VueAwesomeSwiper from 'vue-awesome-swiper'
import { BContainer, BButton } from 'bootstrap-vue'
import _cloneDeep from 'lodash.clonedeep'

import BlockFeaturedPosts from '@/components/blocks/BlockFeaturedPosts'

import createFactory from '@/api/helpers/factory'
import postAttrs from '@/api/factories/post'

const localVue = createLocalVue()

localVue.use(VueAwesomeSwiper)
localVue.component('BContainer', BContainer)
localVue.component('BButton', BButton)

describe.skip('BlockFeaturedPosts', () => {
  let featuredPosts

  beforeEach(() => {
    featuredPosts = _cloneDeep(createFactory(postAttrs, 4))
  })

  test('renders a slider', () => {
    const wrapper = mount(BlockFeaturedPosts, {
      localVue,
      propsData: {
        posts: featuredPosts
      },
      stubs: ['svg-icon', 'NuxtLink'],
      mocks: {
        $i18n: { path: (url) => url },
        $t: (path) => path
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
