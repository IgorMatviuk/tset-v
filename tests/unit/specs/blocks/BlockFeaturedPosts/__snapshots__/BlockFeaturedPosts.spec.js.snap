// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockFeaturedPosts renders a slider 1`] = `
<div
  class="posts"
>
  <div
    class="container"
  >
    <div
      class="section-header"
    >
      <div
        class="section-header__title"
      >
        блог
      </div>
       
      <div
        class="section-header__navigation"
      >
        <div
          class="posts__navigation"
        >
          <div
            class="post__prev swiper-button-prev"
          >
            <svg-icon-stub
              class="swiper-button-prev-icon"
              height="9"
              name="arrow"
              width="30"
            />
          </div>
           
          <div
            class="post__next swiper-button-next"
          >
            <svg-icon-stub
              class="swiper-button-next-icon"
              height="9"
              name="arrow"
              width="30"
            />
          </div>
        </div>
         
        <nuxtlink-stub
          class="more-link posts__more"
          to="blog"
        >
          
          Весь блог
         
          <svg-icon-stub
            class="more-link__icon"
            height="9"
            name="caret"
            width="11"
          />
        </nuxtlink-stub>
      </div>
    </div>
     
    <div
      class="posts__slider-container"
    >
      <div
        class="posts__slider swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-wrapper posts__slider-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide posts__slide"
          >
            <div
              class="post-card"
            >
              <div
                class="post-card__row"
              >
                <div
                  class="post-card__preview"
                >
                  <div
                    class="post-card__img-wrapper"
                  >
                    <div
                      class="img-loader img-loader--negative"
                    >
                      <picture>
                        <source
                          data-srcset="/img/featured-posts/mock-1__xxl.webp"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xxl.jpg"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xl.webp"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xl.jpg"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__lg.webp"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__lg.jpg"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__sm.webp"
                          type="image/webp"
                        />
                         
                        <img
                          alt="Совет в декрет! Как подобрать стильные летние платья и сарафан"
                          class="post-card__img lazyload img-fade"
                          data-src="/img/featured-posts/mock-1__sm.jpg"
                          loading="lazy"
                          src="/img/stub.png"
                          title="Совет в декрет! Как подобрать стильные летние платья и сарафан"
                        />
                      </picture>
                    </div>
                     
                    <div
                      class="post-card__img-overlay"
                    />
                     
                    <div
                      class="post-card__button-group"
                    >
                      <a
                        class="btn post-card__button is-blog btn-outline-light"
                        href="/"
                        target="_self"
                      >
                        
            в блог
          
                      </a>
                    </div>
                  </div>
                </div>
                 
                <article
                  class="post-card__content"
                >
                  <div
                    class="post-card__date"
                  >
                    17.07.2018
                  </div>
                   
                  <h3
                    class="post-card__title"
                  >
                    Совет в декрет! Как подобрать стильные летние платья и сарафан
                  </h3>
                   
                  <p
                    class="post-card__post"
                  >
                    В самое волнующее и прекрасное для всех девушек время хочется чувствовать себя очень комфортно и уютно. Чтобы за эти 9 месяцев можно было насладиться жизнью, ощущать и получать только крутые впечатления. А выглядеть хочется так же красиво и женственно. Ведь растянутые футболки и лосины с вытянутыми коленками – это только для дома. Ведь представляешь, лето, жара, солнце, а ты в красивом и удобном образе
                  </p>
                   
                  <nuxtlink-stub
                    class="more-link post-card__link more-link--is-black"
                    to="some-article"
                  >
                    
        buttons.readMore
       
                    <svg-icon-stub
                      class="more-link__icon"
                      height="9"
                      name="caret"
                      width="11"
                    />
                  </nuxtlink-stub>
                </article>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide posts__slide"
          >
            <div
              class="post-card"
            >
              <div
                class="post-card__row"
              >
                <div
                  class="post-card__preview"
                >
                  <div
                    class="post-card__img-wrapper"
                  >
                    <div
                      class="img-loader img-loader--negative"
                    >
                      <picture>
                        <source
                          data-srcset="/img/featured-posts/mock-2__xxl.webp"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xxl.jpg"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xl.webp"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xl.jpg"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__lg.webp"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__lg.jpg"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__sm.webp"
                          type="image/webp"
                        />
                         
                        <img
                          alt="Самые трендовые принты лета 2019"
                          class="post-card__img lazyload img-fade"
                          data-src="/img/featured-posts/mock-2__sm.jpg"
                          loading="lazy"
                          src="/img/stub.png"
                          title="Самые трендовые принты лета 2019"
                        />
                      </picture>
                    </div>
                     
                    <div
                      class="post-card__img-overlay"
                    />
                     
                    <div
                      class="post-card__button-group"
                    >
                      <a
                        class="btn post-card__button is-blog btn-outline-light"
                        href="/"
                        target="_self"
                      >
                        
            в блог
          
                      </a>
                    </div>
                  </div>
                </div>
                 
                <article
                  class="post-card__content"
                >
                  <div
                    class="post-card__date"
                  >
                    12.07.2018
                  </div>
                   
                  <h3
                    class="post-card__title"
                  >
                    Самые трендовые принты лета 2019
                  </h3>
                   
                  <p
                    class="post-card__post"
                  >
                    Наступившее лето — отличный повод порадовать себя обновками. Летние платья, блузы, сарафаны никогда не бывают лишними в гардеробе, а вот выбрать их расцветки с учетом самых последних трендов в дизайне поможет эта статья с обзором модных принтов.  Одним из самых трендовых принтов весенне-летнего сезона 2019 стал принт «Tie Dye», название которого можно перевести как «завяжи и покрась». Оттенки самые разные — от
                  </p>
                   
                  <nuxtlink-stub
                    class="more-link post-card__link more-link--is-black"
                    to="some-article"
                  >
                    
        buttons.readMore
       
                    <svg-icon-stub
                      class="more-link__icon"
                      height="9"
                      name="caret"
                      width="11"
                    />
                  </nuxtlink-stub>
                </article>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide posts__slide"
          >
            <div
              class="post-card"
            >
              <div
                class="post-card__row"
              >
                <div
                  class="post-card__preview"
                >
                  <div
                    class="post-card__img-wrapper"
                  >
                    <div
                      class="img-loader img-loader--negative"
                    >
                      <picture>
                        <source
                          data-srcset="/img/featured-posts/mock-1__xxl.webp"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xxl.jpg"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xl.webp"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__xl.jpg"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__lg.webp"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__lg.jpg"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-1__sm.webp"
                          type="image/webp"
                        />
                         
                        <img
                          alt="Совет в декрет! Как подобрать стильные летние платья и сарафан"
                          class="post-card__img lazyload img-fade"
                          data-src="/img/featured-posts/mock-1__sm.jpg"
                          loading="lazy"
                          src="/img/stub.png"
                          title="Совет в декрет! Как подобрать стильные летние платья и сарафан"
                        />
                      </picture>
                    </div>
                     
                    <div
                      class="post-card__img-overlay"
                    />
                     
                    <div
                      class="post-card__button-group"
                    >
                      <a
                        class="btn post-card__button is-blog btn-outline-light"
                        href="/"
                        target="_self"
                      >
                        
            в блог
          
                      </a>
                    </div>
                  </div>
                </div>
                 
                <article
                  class="post-card__content"
                >
                  <div
                    class="post-card__date"
                  >
                    17.07.2018
                  </div>
                   
                  <h3
                    class="post-card__title"
                  >
                    Совет в декрет! Как подобрать стильные летние платья и сарафан
                  </h3>
                   
                  <p
                    class="post-card__post"
                  >
                    В самое волнующее и прекрасное для всех девушек время хочется чувствовать себя очень комфортно и уютно. Чтобы за эти 9 месяцев можно было насладиться жизнью, ощущать и получать только крутые впечатления. А выглядеть хочется так же красиво и женственно. Ведь растянутые футболки и лосины с вытянутыми коленками – это только для дома. Ведь представляешь, лето, жара, солнце, а ты в красивом и удобном образе
                  </p>
                   
                  <nuxtlink-stub
                    class="more-link post-card__link more-link--is-black"
                    to="some-article"
                  >
                    
        buttons.readMore
       
                    <svg-icon-stub
                      class="more-link__icon"
                      height="9"
                      name="caret"
                      width="11"
                    />
                  </nuxtlink-stub>
                </article>
              </div>
            </div>
          </div>
          <div
            class="swiper-slide posts__slide"
          >
            <div
              class="post-card"
            >
              <div
                class="post-card__row"
              >
                <div
                  class="post-card__preview"
                >
                  <div
                    class="post-card__img-wrapper"
                  >
                    <div
                      class="img-loader img-loader--negative"
                    >
                      <picture>
                        <source
                          data-srcset="/img/featured-posts/mock-2__xxl.webp"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xxl.jpg"
                          media="(min-width: 1500px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xl.webp"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__xl.jpg"
                          media="(min-width: 1200px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__lg.webp"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/webp"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__lg.jpg"
                          media="(min-width: 992px)"
                          srcset="/img/stub.png"
                          type="image/jpeg"
                        />
                         
                        <source
                          data-srcset="/img/featured-posts/mock-2__sm.webp"
                          type="image/webp"
                        />
                         
                        <img
                          alt="Самые трендовые принты лета 2019"
                          class="post-card__img lazyload img-fade"
                          data-src="/img/featured-posts/mock-2__sm.jpg"
                          loading="lazy"
                          src="/img/stub.png"
                          title="Самые трендовые принты лета 2019"
                        />
                      </picture>
                    </div>
                     
                    <div
                      class="post-card__img-overlay"
                    />
                     
                    <div
                      class="post-card__button-group"
                    >
                      <a
                        class="btn post-card__button is-blog btn-outline-light"
                        href="/"
                        target="_self"
                      >
                        
            в блог
          
                      </a>
                    </div>
                  </div>
                </div>
                 
                <article
                  class="post-card__content"
                >
                  <div
                    class="post-card__date"
                  >
                    12.07.2018
                  </div>
                   
                  <h3
                    class="post-card__title"
                  >
                    Самые трендовые принты лета 2019
                  </h3>
                   
                  <p
                    class="post-card__post"
                  >
                    Наступившее лето — отличный повод порадовать себя обновками. Летние платья, блузы, сарафаны никогда не бывают лишними в гардеробе, а вот выбрать их расцветки с учетом самых последних трендов в дизайне поможет эта статья с обзором модных принтов.  Одним из самых трендовых принтов весенне-летнего сезона 2019 стал принт «Tie Dye», название которого можно перевести как «завяжи и покрась». Оттенки самые разные — от
                  </p>
                   
                  <nuxtlink-stub
                    class="more-link post-card__link more-link--is-black"
                    to="some-article"
                  >
                    
        buttons.readMore
       
                    <svg-icon-stub
                      class="more-link__icon"
                      height="9"
                      name="caret"
                      width="11"
                    />
                  </nuxtlink-stub>
                </article>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="swiper-pagination posts__pagination"
        />
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
    </div>
  </div>
</div>
`;
